packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -375,25 +367,21 @@ InitializeResult ArbCore::applyConfig() {
 
     if (coreConfig.database_exit_after) {
         // Exit program, nothing else to do.
-        if (!status.ok()) {
-            std::cerr << "Exiting early, but error occurred: "
-                      << status.ToString() << std::endl;
-        }
-
         return {rocksdb::Status::OK(), true};
     }
+
     if (!reorg_done) {
-        status = reorgToLastMessage(cache);
+        return {reorgToLastMessage(cache), false};
     }
 
-    return {status, false};
+    return {rocksdb::Status::OK(), false};
 }
 
 InitializeResult ArbCore::initialize(const LoadedExecutable& executable) {
     // Use latest existing checkpoint
 
     auto config_result = applyConfig();
-    if (config_result.finished) {
+    if (config_result.finished || config_result.status.ok()) {
```

Comment:
Should this be `!config_result.status.ok()`?

Reply:
This is actually the intended.  We only need to continue with initialize if applyConfig returns an error which means that the database isn't properly setup.